cairo/src/utils/bytes.cairo

```diff
@@ -288,77 +288,132 @@ func bytes_to_bytes8_little_endian{range_check_ptr}(dst: felt*, bytes_len: felt,
         return ();
     }
 
-    let (local pow256) = get_label_location(pow256_table);
-    let (full_u64_word_count, local last_input_num_bytes) = unsigned_div_rem(bytes_len, 8);
-    local range_check_ptr = range_check_ptr;
-
-    tempvar dst_index = 0;
-    tempvar bytes_index = bytes_len - 1;
-    tempvar bytes8 = 0;
-    tempvar bytes8_index = 7;
-
-    body:
-    let dst_index = [ap - 4];
-    let bytes_index = [ap - 3];
-    let bytes8 = [ap - 2];
-    let bytes8_index = [ap - 1];
-
-    let bytes_len = [fp - 4];
-    let bytes = cast([fp - 3], felt*);
-    let pow256 = cast([fp], felt*);
-    let current_byte = bytes[bytes_len - 1 - bytes_index];
-    let current_pow = pow256[bytes8_index];
+    tempvar less_than_8;
+    %{ ids.less_than_8 = int(ids.bytes_len < 8) %}
```

Comment:
why do we do these operations inside hints? they look inexpensive

also, what happens if the prover 'lies' here? e.g. bytes_len = 10 but the prover writes `ids.less_than_8 = int(True)`